jquery.flot.touchNavigate.js

```diff
@@ -14,175 +14,369 @@
-                    prevPanY = e.touches[0].pageY;
+            },
+
+            actionOnDragEnd: function(e) {
+                prevDist = null;
+                if (wasPinchEvent(e)) {
+                    prevPan.x = getPoint(e).x;
+                    prevPan.y = getPoint(e).y;
                 }
             }
         }
 
-        function onDragEnd(e) {
-            if (!isPinchEvent(e)) {
+        var zoomPinchOnAxis = {
+            conditionOnStart: function(e) {
+                return twoTouches && setAxisDoubleTouch(e) && !noAxisTouched();
+            },
+
+            conditionOnDrag: function(e) {
+                return (twoTouches && !noAxisTouched());
+            },
+
+            conditionOnDragEnd: function(e) {
+                return !isPinchEvent(e);
+            },
+
+            actionOnStart: function(e) {
+                preventEventPropagation(e);
+                setPrevDist(e);
+                updateAxisData(e, getMidPoint(e));
+            },
+
+            actionOnDrag: function(e) {
+                var dist = pinchDistance(e);
+                onZoomPinch(e);
+                prevDist = dist;
+            },
+
+            actionOnDragEnd: function(e) {
                 prevDist = null;
-                if (twoTouches && e.touches.length === 1) {
-                    prevPanX = e.touches[0].pageX;
-                    prevPanY = e.touches[0].pageY;
-                } else if (!twoTouches) {
-                    var currentTime = new Date().getTime(),
-                        intervalBetweenTaps = currentTime - prevTapTime,
-                        maxDistanceBetweenTaps = 20,
-                        maxIntervalBetweenTaps = 500;
-
-                    if (intervalBetweenTaps >= 0 && intervalBetweenTaps < maxIntervalBetweenTaps) {
-                        if ((xAxisTouched && prevAxisTouched === 'xaxis' && (Math.abs(prevTapX - prevPanX) < maxDistanceBetweenTaps)) ||
-                            (yAxisTouched && prevAxisTouched === 'yaxis' && (Math.abs(prevTapY - prevPanY) < maxDistanceBetweenTaps)) ||
-                            (!xAxisTouched && prevAxisTouched === 'none' && !yAxisTouched && distance(prevTapX, prevTapY, prevPanX, prevPanY) < maxDistanceBetweenTaps)) {
-                            plot.recenter({ axes: axis });
-                        }
-                    }
-                    if (xAxisTouched) {
-                        prevAxisTouched = 'xaxis';
-                    } else if (yAxisTouched) {
-                        prevAxisTouched = 'yaxis';
-                    } else prevAxisTouched = 'none';
-                    prevTapTime = currentTime;
+                if (wasPinchEvent(e)) {
+                    prevPan.x = getPoint(e).x;
+                    prevPan.y = getPoint(e).y;
                 }
             }
         }
 
+        var gestures = [oneTouchPan, doubleTouchPan, oneTouchPanOnAxis, doubleTouchPanOnAxis, zoomPinch, zoomPinchOnAxis];
```

Comment:
I'd like to see the gestures and actions separate, not mixed.

The gestures that make sense, in my opinion, are "tap", "doubleTap", "pan" and "zoom:

The function that deal with these need to have access to information like: 
* first tap position
* previous tap position
* two position deltas (one relative to the first touch, the one relative to the previous touch)
..... plus a lot of infor related to zoom in vase of zoom events